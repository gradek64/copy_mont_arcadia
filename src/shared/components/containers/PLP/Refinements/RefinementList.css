/** @define RefinementList */
.RefinementList {
  overflow: auto;
  flex-grow: 2;
  position: relative;
  margin-top: -1px; /* Hide the top Accordion border when user is scrolled to top */

  &-accordionHeader {
    display: flex;
    flex-direction: row;
    font-weight: $refinements-items-header-font-weight;
  }

  &-label {
    min-height: $refinements-items-header-height;

    &.is-rating::after {
      display: $refinements-item-rating-display-selection;
    }
  }

  &-selection {
    margin-left: $refinements-items-selection-margin-left;
    flex-grow: 1;
    font-size: $refinements-items-selection-font-size;
    line-height: $refinements-items-selection-line-height;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;

    &.is-rating {
      display: $refinements-item-rating-display-selection;
    }
  }
}

/* postcss-bem-linter: ignore */
.RefinementList .Accordion {
  border-color: $refinements-accordion-border-color;
}

@media $min-tablet {
  .RefinementList {
    overflow: initial;
  }
}

