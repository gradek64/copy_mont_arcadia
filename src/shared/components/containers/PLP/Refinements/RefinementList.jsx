import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import RefinementOptions from './RefinementOptions'
import { capitalize } from '../../../../lib/string-utils'
import * as actions from '../../../../actions/components/refinementsActions'
import Accordion from '../../../common/Accordion/Accordion'
import AccordionGroup from '../../../common/AccordionGroup/AccordionGroup'
import Price from '../../../common/Price/Price'
import { isMobile } from '../../../../selectors/viewportSelectors'
import {
  getRefinements,
  getActiveRefinementsState,
} from '../../../../selectors/refinementsSelectors'
import { pathOr } from 'ramda'
import { analyticsPlpClickEvent } from '../../../../analytics/tracking/site-interactions'
import { getCategoryTitle } from '../../../../selectors/productSelectors'

const header = (isMobile, label, type, selectionText) => (
  <div
    role="presentation"
    className="RefinementList-accordionHeader"
    onClick={() => {
      analyticsPlpClickEvent(`${label.toLowerCase()}`)
    }}
  >
    <span
      className={`RefinementList-label ${
        selectionText ? 'is-filtered' : ''
      } is-${type.toLowerCase()}`}
    >
      {label}
    </span>
    {isMobile && (
      <span className={`RefinementList-selection is-${type.toLowerCase()}`}>
        {selectionText}
      </span>
    )}
  </div>
)

@connect(
  (state) => ({
    refinements: getRefinements(state),
    selectedOptions: state.refinements.selectedOptions,
    isMobile: isMobile(state),
    productCategory: getCategoryTitle(state),
    getActiveRefinements: getActiveRefinementsState(state),
  }),
  { ...actions }
)
class RefinementList extends Component {
  static propTypes = {
    selectedOptions: PropTypes.object,
    refinements: PropTypes.array,
    isMobile: PropTypes.bool,
    onAccordionToggle: PropTypes.func,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  static defaultProps = {
    onAccordionToggle: () => {},
    selectedOptions: {},
    refinements: [],
    isMobile: true,
  }

  setRefinementsAppliedForNoFollow(refinements) {
    let refinementsLength = refinements.length
    if (refinements.find((ref) => ref.propertyName === 'nowPrice')) {
      refinementsLength -= 1
    }
    if (
      refinements.find(
        (ref) => ref.dimensionName && ref.dimensionName.includes('category')
      )
    ) {
      refinementsLength -= 1
    }
    return refinementsLength
  }

  setRefinementType(options, type) {
    const rating = type === 'Rating' ? 'RATING' : ''
    const size = type.includes('Size') ? 'SIZE' : ''
    const firstOption = options[0] ? options[0].type : ''
    return rating || size || firstOption
  }

  selectionText(selections, type) {
    if (!selections || !selections.length) return ''

    const { l } = this.context
    switch (type) {
      case 'VALUE':
        return selections.join(', ').replace(/\w\S*/g, capitalize)
      case 'SIZE':
        return selections.join(', ').toUpperCase()
      case 'RATING': {
        const rating = selections[0].toString()
        return rating === '5'
          ? `${rating} ${l`stars`}`
          : `${rating} ${l`stars`} ${l`& up`}`
      }
      case 'RANGE':
        return (
          <div>
            <Price price={selections[0]} /> - <Price price={selections[1]} />
          </div>
        )
      default:
        return selections.join(' - ').replace(/\w\S*/g, capitalize)
    }
  }

  returnInitiallyExpanded(isMobile, refinements) {
    // items without refinementOptions are ignored
    // RANGE refinements are always expanded.
    return isMobile
      ? []
      : refinements
          .filter(
            (refinement) =>
              refinement.refinementOptions &&
              refinement.refinementOptions.length
          )
          .reduce(
            (expanded, { refinementOptions, label } = {}) =>
              refinementOptions[0].type === 'RANGE' ||
              refinementOptions.some(({ selectedFlag }) => selectedFlag)
                ? [...expanded, label]
                : expanded,
            []
          )
  }

  parseSelectedRefinements(refinementOptions, selectedOptions) {
    return refinementOptions.reduce((selectedRefinements, option) => {
      const selected = pathOr(null, ['price'], selectedOptions)
      if (option.type === 'RANGE' && selected) {
        return [
          ...selectedRefinements,
          parseInt(selected[0], 10),
          parseInt(selected[1], 10),
        ]
      }
      return option.selectedFlag
        ? [...selectedRefinements, option.label]
        : selectedRefinements
    }, [])
  }

  blockRefinementItem = (type, refinementOptions) => {
    switch (type) {
      case '':
      case undefined:
        return true
      case 'RANGE': {
        const [{ minValue, maxValue }] = refinementOptions
        return Number(minValue) >= Number(maxValue)
      }
      default:
        return false
    }
  }

  renderRefinementItem = (refineObj) => {
    const { isMobile, selectedOptions, getActiveRefinements } = this.props
    const { refinementOptions, label } = refineObj
    const type = this.setRefinementType(refinementOptions, label)
    const selection = this.parseSelectedRefinements(
      refinementOptions,
      selectedOptions
    )
    const selectionText = refinementOptions.length
      ? this.selectionText(selection, type)
      : ''

    return this.blockRefinementItem(type, refinementOptions) ? null : (
      <Accordion
        key={label}
        header={header(isMobile, label, type, selectionText)}
        accordionName={label}
        scrollPaneSelector=".RefinementList"
        noContentPadding={isMobile}
        noContentBorderTop={!isMobile}
        label={label}
        noHeaderPadding={!isMobile}
        noExpandedHeaderBackground
        arrowStyle="secondary"
        arrowPosition="right"
        className={`Accordion--${label}`} // TODO: EXP-373 Remove this after test has ran
      >
        <RefinementOptions
          options={refinementOptions}
          type={type}
          label={label}
          selections={selection}
          isMobile={isMobile}
          activeRefinements={this.setRefinementsAppliedForNoFollow(
            getActiveRefinements
          )}
        />
      </Accordion>
    )
  }

  render() {
    const {
      onAccordionToggle,
      isMobile,
      refinements,
      handleCollapseAll,
      productCategory,
    } = this.props

    return (
      <AccordionGroup
        className="RefinementList"
        groupName="refinements"
        singleOpen={isMobile}
        initiallyExpanded={this.returnInitiallyExpanded(isMobile, refinements)}
        onAccordionToggle={onAccordionToggle}
        handleCollapseAll={handleCollapseAll}
        productCategory={productCategory}
      >
        {refinements.map(this.renderRefinementItem).filter(Boolean)}
      </AccordionGroup>
    )
  }
}

export default RefinementList
