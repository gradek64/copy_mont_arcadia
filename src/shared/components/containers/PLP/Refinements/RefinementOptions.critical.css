/** @define RefinementOptions */
.RefinementOptions {
  display: block;
  margin: 0;
  padding-left: 0;
  list-style: none;
}
