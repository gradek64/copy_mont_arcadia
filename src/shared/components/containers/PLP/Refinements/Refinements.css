/** @define Refinements */
.Refinements {
  position: fixed;
  z-index: -1;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  visibility: hidden;
  transition: all 0.3s ease;
  opacity: 0;
  background: rgba(51, 51, 51, 0.8);

  &.is-shown {
    visibility: visible;
    z-index: 10;
    opacity: 1;
  }

  &-content {
    display: flex;
    flex-direction: column;
    position: absolute;
    top: $refinements-margin;
    left: $refinements-margin;
    right: $refinements-margin;
    max-height: calc(100% - 2 * $refinements-margin);
    transform: none;
    animation-duration: 0.3s;
    background: $refinements-background-color;

    &.is-shown {
      animation-name: zoomIn;
    }
    &.is-hidden {
      animation-name: zoomOut;
    }
  }

  &-header {
    position: relative;
    box-sizing: border-box;
    border-bottom: $refinements-header-border-bottom;
    background: $refinements-header-background-color;
    padding: $refinements-header-padding;
    z-index: 2;
    flex-shrink: 0;
  }

  &-title {
    text-align: center;
    margin-top: $refinements-title-margin-top;
    margin-bottom: $refinements-title-margin-bottom;
    font-size: $refinements-title-font-size;
    font-weight: $refinements-title-font-weight;
    color: $refinements-title-color;
  }

  &-closeIcon {
    position: absolute;
    top: $refinements-close-icon-top;
    right: $refinements-close-icon-right;
    width: 40px;
    height: 40px;
    background: url('public/$brand/images/refinements-close-icon.svg') 50%/$refinements-close-icon-width no-repeat;
  }

  &-clearButton {
    text-decoration: $refinements-clear-button-text-decoration;
    font-size: $refinements-clear-button-font-size;
    color: $refinements-clear-button-color;
    padding: $refinements-clear-button-padding;
    float: right;
    position: $refinements-clear-button-position;
    top: $refinements-clear-button-top;
    left: $refinements-clear-button-left;
    text-transform: $refinements-clear-button-text-transform;
    transition: color 0.1s linear;

    &:disabled {
      text-decoration: $refinements-clear-button-disabled-text-decoration;
      padding: $refinements-clear-button-disabled-padding;
      color: $refinements-clear-button-disabled-color;

      &::before {
        opacity: $refinements-clear-button-disabled-icon-opacity;
      }
    }

    &::before {
      content: '';
      display: inline-block;
      background-image: $refinements-clear-button-icon-image;
      width: $refinements-clear-button-icon-width;
      height: $refinements-clear-button-icon-height;
      margin-right: $refinements-clear-button-icon-margin-right;
      background-repeat: no-repeat;
      background-position: center;
      background-size: contain;
      vertical-align: text-top;
      transition: opacity 0.1s linear;
    }
  }

  &-applyButtonWrapper {
    padding: $refinements-apply-button-padding;
    border-top: $refinements-apply-button-border-top;
  }
  &-applyButton {
    background-color: $refinements-apply-button-background-color;
    border-color: $refinements-apply-button-border-color;
  }

  &-loader {
    display: block;
    left: 0;
  }
}

/* postcss-bem-linter: ignore */
.ios .Refinements-content {
  max-height: calc(100% - (2 * $refinements-margin) - 20px);
}
