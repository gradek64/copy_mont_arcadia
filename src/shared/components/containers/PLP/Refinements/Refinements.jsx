import PropTypes from 'prop-types'
import React, { Component } from 'react'
import { connect } from 'react-redux'

import { isMobile } from '../../../../selectors/viewportSelectors'

// actions
import * as actions from '../../../../actions/components/refinementsActions'
import { sendAnalyticsFilterUsedEvent } from '../../../../analytics/analytics-actions'

// components
import Button from '../../../common/Button/Button'
import AccessibleText from '../../../common/AccessibleText/AccessibleText'
import RefinementList from './RefinementList'
import {
  isRefinementsSelected,
  getRemoveAllFilters,
} from '../../../../../shared/selectors/refinementsSelectors'
import LoaderOverlay from '../../../common/LoaderOverlay/LoaderOverlay'
import EnhancedMessage from '../../../common/EnhancedMessage/EnhancedMessage'

// Helpers
import {
  isSeoUrlSearchFilter,
  getResetSearchUrl,
  isSearchUrl,
  addParamClearAll,
} from '../../../../lib/products-utils'
import {
  getRouteSearch,
  getRoutePath,
} from '../../../../selectors/routingSelectors'
import { getProductsSearchResultsTotal } from '../../../../selectors/productSelectors'

@connect(
  (state) => ({
    isRefinementsSelected: isRefinementsSelected(state),
    removeAllRefinement: getRemoveAllFilters(state),
    isShown: state.refinements.isShown,
    isMobile: isMobile(state),
    isLoadingRefinements: state.refinementsV2.isLoadingRefinements,
    currentPath: getRoutePath(state),
    currentSearchPath: getRouteSearch(state),
    totalProducts: getProductsSearchResultsTotal(state),
  }),
  {
    ...actions,
    sendAnalyticsFilterUsedEvent,
    getRemoveAllFilters,
  }
)
class Refinements extends Component {
  static propTypes = {
    isRefinementsSelected: PropTypes.bool,
    removeAllRefinement: PropTypes.string,
    isShown: PropTypes.bool,
    isMobile: PropTypes.bool,
    isLoadingRefinements: PropTypes.bool,
    currentPath: PropTypes.string,
    currentSearchPath: PropTypes.string,
    toggleRefinements: PropTypes.func,
    updateRefinements: PropTypes.func,
    applyRefinementsMobile: PropTypes.func,
    resetRefinements: PropTypes.func,
    sendAnalyticsFilterUsedEvent: PropTypes.func,
    totalProducts: PropTypes.number,
  }

  static contextTypes = {
    l: PropTypes.func,
  }

  static defaultProps = {
    isMobile: true,
  }

  componentDidUpdate({ isShown: wasShown }) {
    const { isShown } = this.props
    if (isShown && !wasShown && this.description) this.description.focus()
  }

  componentWillUnmount() {
    const { isMobile } = this.props
    if (isMobile) {
      this.closeRefinementsHandler()
    }
  }

  closeRefinementsHandler = () => {
    const { toggleRefinements, resetRefinements } = this.props

    resetRefinements()
    toggleRefinements(false)

    if (this.collapseAllHandler) {
      this.collapseAllHandler()
    }
  }

  applyRefinementsHandler = () => {
    const {
      toggleRefinements,
      applyRefinementsMobile,
      sendAnalyticsFilterUsedEvent,
    } = this.props

    applyRefinementsMobile()
    if (this.collapseAllHandler) {
      this.collapseAllHandler()
    }
    toggleRefinements(false)

    sendAnalyticsFilterUsedEvent({
      filterCategory: 'apply filters',
      filterOption: 'apply filters',
      filterAction: 'click',
    })
  }

  clearAndApplyRefinements = () => {
    const {
      sendAnalyticsFilterUsedEvent,
      removeAllRefinement,
      updateRefinements,
      currentPath,
      currentSearchPath,
    } = this.props

    if (isSeoUrlSearchFilter(removeAllRefinement)) {
      if (isSearchUrl(currentPath)) {
        updateRefinements(`${currentPath}${currentSearchPath}`)
      } else {
        updateRefinements(getResetSearchUrl(currentSearchPath))
      }
    } else {
      updateRefinements(addParamClearAll(removeAllRefinement))
    }
    // We using the old refinement reducer for resetting priceRange
    // This is needed for Slider Component (TODO : improve this component/logic)
    this.props.removeOptionRange('price')
    if (this.collapseAllHandler) {
      this.collapseAllHandler()
    }

    sendAnalyticsFilterUsedEvent({
      filterCategory: 'clear all',
      filterOption: 'clear all',
      filterAction: 'remove',
    })
  }

  renderNoResultsFound = () => {
    const { l } = this.context
    const { isMobile, totalProducts } = this.props

    if (!isMobile) {
      return
    }

    if (!totalProducts) {
      return (
        <EnhancedMessage
          message={l`There are no products matching your current search criteria.`}
          isFromFilter
          renderChildrenOn="bottom"
        />
      )
    }
  }

  renderHeader = () => {
    const { l } = this.context
    const { isRefinementsSelected } = this.props
    return (
      <div className="Refinements-header">
        <h3 className="Refinements-title">{l`Filter`}</h3>
        <AccessibleText
          ref={(accessibleText) => {
            this.description = accessibleText
          }}
        >
          {l`This is the filters modal. Select any required filters and press apply to return to a filtered product list.`}
        </AccessibleText>
        <button
          className="Refinements-closeIcon"
          onClick={this.closeRefinementsHandler}
        >
          <span className="screen-reader-text">{l`Close filters modal`}</span>
        </button>
        <button
          className="Refinements-clearButton"
          onClick={this.clearAndApplyRefinements}
          disabled={!isRefinementsSelected}
        >
          {l`Clear all`}
        </button>
      </div>
    )
  }

  handleCollapseAll = (cb) => {
    this.collapseAllHandler = cb
  }

  render() {
    const {
      isShown,
      isMobile,
      isLoadingRefinements,
      totalProducts,
    } = this.props
    const { l } = this.context

    return isMobile ? (
      <div className={`Refinements ${isShown ? 'is-shown' : 'is-hidden'}`}>
        <div
          className={`Refinements-content ${
            isShown ? 'is-shown' : 'is-hidden'
          }`}
        >
          {this.renderHeader()}
          <RefinementList handleCollapseAll={this.handleCollapseAll} />
          {this.renderNoResultsFound()}
          <div className="Refinements-applyButtonWrapper">
            <Button
              isDisabled={!totalProducts}
              className="Refinements-applyButton"
              clickHandler={this.applyRefinementsHandler}
            >
              {l`Apply`}
            </Button>
          </div>
          {isLoadingRefinements && (
            <LoaderOverlay className="Refinements-loader" />
          )}
        </div>
      </div>
    ) : null
  }
}

export default Refinements
