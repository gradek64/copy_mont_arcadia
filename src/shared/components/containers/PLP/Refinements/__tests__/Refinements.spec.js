import testComponentHelper from 'test/unit/helpers/test-component'
import PropTypes from 'prop-types'
import React from 'react'
import { mount } from 'enzyme'
import { createStore } from 'redux'
import Refinements from '../Refinements'

describe('<RefinementList/>', () => {
  jest.mock('../../../../../lib/products-utils', () => ({
    isSeoUrlSearchFilter: jest.fn(
      (seoUrl) => seoUrl && seoUrl.includes('seo=false')
    ),
    isSearchUrl: jest.fn((seoUrl) => seoUrl && seoUrl.includes('/search/')),
    getResetSearchUrl: jest.fn(
      (currentSearchPath) =>
        `/search/?q=${currentSearchPath.match(/Ntt=(.*?)&/)[1]}`
    ),
  }))
  const renderComponent = testComponentHelper(Refinements.WrappedComponent)
  const refinements = [
    {
      label: 'Colour',
      refinementOptions: [
        {
          type: 'VALUE',
          label: 'black',
          value: 'black',
        },
      ],
    },
    {
      label: 'Size',
      refinementOptions: [
        {
          type: 'VALUE',
          label: 4,
          value: 4,
        },
        {
          type: 'VALUE',
          label: 6,
          value: 6,
        },
      ],
    },
    {
      label: 'Price',
      refinementOptions: [
        {
          type: 'RANGE',
          label: 0,
          value: 0,
        },
      ],
    },
  ]
  const defaultProps = {
    refinements: [],
  }

  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(defaultProps).getTree()).toMatchSnapshot()
    })
    it('with refinements', () => {
      expect(
        renderComponent({ ...defaultProps, refinements }).getTree()
      ).toMatchSnapshot()
    })
    it('with refinements and selectedOptions', () => {
      const selectedOptions = {
        Size: [4],
      }
      expect(
        renderComponent({
          ...defaultProps,
          numRefinements: 1,
          refinements,
          selectedOptions,
        }).getTree()
      ).toMatchSnapshot()
    })
    it('with isShown', () => {
      expect(
        renderComponent({ ...defaultProps, isShown: true }).getTree()
      ).toMatchSnapshot()
    })
    it('with location', () => {
      expect(
        renderComponent({
          ...defaultProps,
          location: {
            pathname: 'some/pathname',
          },
        }).getTree()
      ).toMatchSnapshot()
    })
    it('with isMobile as false', () => {
      expect(
        renderComponent({ ...defaultProps, isMobile: false }).getTree()
      ).toMatchSnapshot()
    })
    it('renders overlay when loading and featureflag on', () => {
      expect(
        renderComponent({
          ...defaultProps,
          isLoadingRefinements: true,
        }).getTree()
      ).toMatchSnapshot()
    })
  })

  describe('@lifecycle', () => {
    beforeEach(() => {
      jest.resetAllMocks()
    })
    describe('componentDidUpdate', () => {
      it('should focus description', () => {
        const state = {
          config: {
            brandName: 'TS',
            brandCode: 'TS',
          },
          products: {
            refinements: [],
          },
          refinements: {},
          features: {
            status: {},
          },
        }
        const store = createStore((state) => state, state)
        const { WrappedComponent } = Refinements
        const context = {
          l: (value) => {
            return value
          },
          store,
        }
        const wrapper = mount(<WrappedComponent isShown {...defaultProps} />, {
          context,
          childContextTypes: {
            l: PropTypes.any,
            store: PropTypes.any,
          },
        })
        const instance = wrapper.instance()
        instance.description.focus = jest.fn()
        instance.componentDidUpdate({ isShown: false })
        expect(instance.description.focus).toHaveBeenCalled()
      })
    })
    describe('componentWillUnmount', () => {
      it('should call toggleRefinements', () => {
        const { instance } = renderComponent({
          ...defaultProps,
          resetRefinements: jest.fn(),
          toggleRefinements: jest.fn(),
        })
        instance.componentWillUnmount()
        expect(instance.props.toggleRefinements).toHaveBeenCalledTimes(1)
        expect(instance.props.toggleRefinements).toHaveBeenCalledWith(false)
      })
    })
  })

  describe('@events', () => {
    describe('clearAndApplyRefinements', () => {
      const refinementProps = {
        ...defaultProps,
        updateRefinements: jest.fn(),
        sendAnalyticsFilterUsedEvent: jest.fn(),
        removeOptionRange: jest.fn(),
      }
      beforeEach(() => jest.clearAllMocks())

      it('should update refinements with remove all path', () => {
        const removeAllRefinement =
          '/en/tsuk/category/clothing-427/N-82zZdgl?Nrpp=24&siteId=%2F12556&categoryId=null&clearAll=true'
        const { instance, wrapper } = renderComponent({
          ...refinementProps,
          removeAllRefinement,
        })
        wrapper.find('.Refinements-clearButton').simulate('click')
        expect(instance.props.updateRefinements).toHaveBeenCalledTimes(1)
        expect(instance.props.updateRefinements).toHaveBeenCalledWith(
          removeAllRefinement
        )
      })
      it('should update refinements with remove all path in search filter', () => {
        const { instance, wrapper } = renderComponent({
          ...refinementProps,
          removeAllRefinement: '?seo=false&siteId=%2F12556',
          currentPath: '/filter/N-deoZdgl',
          currentSearchPath: '?Nrpp=24&Ntt=red&seo=false&siteId=%2F12556',
        })
        wrapper.find('.Refinements-clearButton').simulate('click')
        expect(instance.props.updateRefinements).toHaveBeenCalledTimes(1)
        expect(instance.props.updateRefinements).toHaveBeenCalledWith(
          '/search/?q=red'
        )
      })
      it('should update refinements with remove all path in initial search path', () => {
        const { instance, wrapper } = renderComponent({
          ...refinementProps,
          removeAllRefinement: '?seo=false&siteId=%2F12556',
          currentPath: '/search/',
          currentSearchPath: '?q=red',
        })
        wrapper.find('.Refinements-clearButton').simulate('click')
        expect(instance.props.updateRefinements).toHaveBeenCalledTimes(1)
        expect(instance.props.updateRefinements).toHaveBeenCalledWith(
          '/search/?q=red'
        )
      })
    })
  })

  describe('@functions', () => {
    describe('applyRefinementsHandler', () => {
      it('should call toggleRefinements and applyRefinementsMobile', () => {
        const { instance } = renderComponent({
          ...defaultProps,
          toggleRefinements: jest.fn(),
          applyRefinementsMobile: jest.fn(),
          sendAnalyticsFilterUsedEvent: jest.fn(),
        })
        instance.applyRefinementsHandler()
        expect(instance.props.toggleRefinements).toHaveBeenCalledTimes(1)
        expect(instance.props.toggleRefinements).toHaveBeenCalledWith(false)
        expect(
          instance.props.sendAnalyticsFilterUsedEvent
        ).toHaveBeenCalledWith({
          filterCategory: 'apply filters',
          filterOption: 'apply filters',
          filterAction: 'click',
        })
        expect(instance.props.applyRefinementsMobile).toHaveBeenCalledTimes(1)
        expect(instance.props.toggleRefinements).toHaveBeenCalledTimes(1)
      })
    })

    describe('closeRefinementsHandler', () => {
      it('should call toggle refinements with false', () => {
        const { instance } = renderComponent({
          ...defaultProps,
          resetRefinements: jest.fn(),
          toggleRefinements: jest.fn(),
        })

        instance.closeRefinementsHandler()
        expect(instance.props.toggleRefinements).toHaveBeenCalledTimes(1)
        expect(instance.props.toggleRefinements).toHaveBeenCalledWith(false)
      })
      it('should call resetRefinements', () => {
        const { instance } = renderComponent({
          ...defaultProps,
          toggleRefinements: jest.fn(),
          resetRefinements: jest.fn(),
        })

        instance.closeRefinementsHandler()
        expect(instance.props.resetRefinements).toHaveBeenCalledTimes(1)
      })
    })
  })
})
