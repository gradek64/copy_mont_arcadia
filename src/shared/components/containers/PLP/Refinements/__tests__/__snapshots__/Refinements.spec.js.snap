// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RefinementList/> @renders in default state 1`] = `
<div
  className="Refinements is-hidden"
>
  <div
    className="Refinements-content is-hidden"
  >
    <div
      className="Refinements-header"
    >
      <h3
        className="Refinements-title"
      >
        Filter
      </h3>
      <AccessibleText>
        This is the filters modal. Select any required filters and press apply to return to a filtered product list.
      </AccessibleText>
      <button
        className="Refinements-closeIcon"
        onClick={[Function]}
      >
        <span
          className="screen-reader-text"
        >
          Close filters modal
        </span>
      </button>
      <button
        className="Refinements-clearButton"
        disabled={true}
        onClick={[Function]}
      >
        Clear all
      </button>
    </div>
    <Connect(RefinementList)
      handleCollapseAll={[Function]}
    />
    <EnhancedMessage
      duration={false}
      isError={false}
      isFromFilter={true}
      isVisible={true}
      message="There are no products matching your current search criteria."
      onTransitionComplete={[Function]}
      renderChildrenOn="bottom"
      showOnce={false}
    />
    <div
      className="Refinements-applyButtonWrapper"
    >
      <Button
        className="Refinements-applyButton"
        clickHandler={[Function]}
        isDisabled={true}
      >
        Apply
      </Button>
    </div>
  </div>
</div>
`;

exports[`<RefinementList/> @renders renders overlay when loading and featureflag on 1`] = `
<div
  className="Refinements is-hidden"
>
  <div
    className="Refinements-content is-hidden"
  >
    <div
      className="Refinements-header"
    >
      <h3
        className="Refinements-title"
      >
        Filter
      </h3>
      <AccessibleText>
        This is the filters modal. Select any required filters and press apply to return to a filtered product list.
      </AccessibleText>
      <button
        className="Refinements-closeIcon"
        onClick={[Function]}
      >
        <span
          className="screen-reader-text"
        >
          Close filters modal
        </span>
      </button>
      <button
        className="Refinements-clearButton"
        disabled={true}
        onClick={[Function]}
      >
        Clear all
      </button>
    </div>
    <Connect(RefinementList)
      handleCollapseAll={[Function]}
    />
    <EnhancedMessage
      duration={false}
      isError={false}
      isFromFilter={true}
      isVisible={true}
      message="There are no products matching your current search criteria."
      onTransitionComplete={[Function]}
      renderChildrenOn="bottom"
      showOnce={false}
    />
    <div
      className="Refinements-applyButtonWrapper"
    >
      <Button
        className="Refinements-applyButton"
        clickHandler={[Function]}
        isDisabled={true}
      >
        Apply
      </Button>
    </div>
    <Connect(LoaderOverlay)
      className="Refinements-loader"
    />
  </div>
</div>
`;

exports[`<RefinementList/> @renders with isMobile as false 1`] = `""`;

exports[`<RefinementList/> @renders with isShown 1`] = `
<div
  className="Refinements is-shown"
>
  <div
    className="Refinements-content is-shown"
  >
    <div
      className="Refinements-header"
    >
      <h3
        className="Refinements-title"
      >
        Filter
      </h3>
      <AccessibleText>
        This is the filters modal. Select any required filters and press apply to return to a filtered product list.
      </AccessibleText>
      <button
        className="Refinements-closeIcon"
        onClick={[Function]}
      >
        <span
          className="screen-reader-text"
        >
          Close filters modal
        </span>
      </button>
      <button
        className="Refinements-clearButton"
        disabled={true}
        onClick={[Function]}
      >
        Clear all
      </button>
    </div>
    <Connect(RefinementList)
      handleCollapseAll={[Function]}
    />
    <EnhancedMessage
      duration={false}
      isError={false}
      isFromFilter={true}
      isVisible={true}
      message="There are no products matching your current search criteria."
      onTransitionComplete={[Function]}
      renderChildrenOn="bottom"
      showOnce={false}
    />
    <div
      className="Refinements-applyButtonWrapper"
    >
      <Button
        className="Refinements-applyButton"
        clickHandler={[Function]}
        isDisabled={true}
      >
        Apply
      </Button>
    </div>
  </div>
</div>
`;

exports[`<RefinementList/> @renders with location 1`] = `
<div
  className="Refinements is-hidden"
>
  <div
    className="Refinements-content is-hidden"
  >
    <div
      className="Refinements-header"
    >
      <h3
        className="Refinements-title"
      >
        Filter
      </h3>
      <AccessibleText>
        This is the filters modal. Select any required filters and press apply to return to a filtered product list.
      </AccessibleText>
      <button
        className="Refinements-closeIcon"
        onClick={[Function]}
      >
        <span
          className="screen-reader-text"
        >
          Close filters modal
        </span>
      </button>
      <button
        className="Refinements-clearButton"
        disabled={true}
        onClick={[Function]}
      >
        Clear all
      </button>
    </div>
    <Connect(RefinementList)
      handleCollapseAll={[Function]}
    />
    <EnhancedMessage
      duration={false}
      isError={false}
      isFromFilter={true}
      isVisible={true}
      message="There are no products matching your current search criteria."
      onTransitionComplete={[Function]}
      renderChildrenOn="bottom"
      showOnce={false}
    />
    <div
      className="Refinements-applyButtonWrapper"
    >
      <Button
        className="Refinements-applyButton"
        clickHandler={[Function]}
        isDisabled={true}
      >
        Apply
      </Button>
    </div>
  </div>
</div>
`;

exports[`<RefinementList/> @renders with refinements 1`] = `
<div
  className="Refinements is-hidden"
>
  <div
    className="Refinements-content is-hidden"
  >
    <div
      className="Refinements-header"
    >
      <h3
        className="Refinements-title"
      >
        Filter
      </h3>
      <AccessibleText>
        This is the filters modal. Select any required filters and press apply to return to a filtered product list.
      </AccessibleText>
      <button
        className="Refinements-closeIcon"
        onClick={[Function]}
      >
        <span
          className="screen-reader-text"
        >
          Close filters modal
        </span>
      </button>
      <button
        className="Refinements-clearButton"
        disabled={true}
        onClick={[Function]}
      >
        Clear all
      </button>
    </div>
    <Connect(RefinementList)
      handleCollapseAll={[Function]}
    />
    <EnhancedMessage
      duration={false}
      isError={false}
      isFromFilter={true}
      isVisible={true}
      message="There are no products matching your current search criteria."
      onTransitionComplete={[Function]}
      renderChildrenOn="bottom"
      showOnce={false}
    />
    <div
      className="Refinements-applyButtonWrapper"
    >
      <Button
        className="Refinements-applyButton"
        clickHandler={[Function]}
        isDisabled={true}
      >
        Apply
      </Button>
    </div>
  </div>
</div>
`;

exports[`<RefinementList/> @renders with refinements and selectedOptions 1`] = `
<div
  className="Refinements is-hidden"
>
  <div
    className="Refinements-content is-hidden"
  >
    <div
      className="Refinements-header"
    >
      <h3
        className="Refinements-title"
      >
        Filter
      </h3>
      <AccessibleText>
        This is the filters modal. Select any required filters and press apply to return to a filtered product list.
      </AccessibleText>
      <button
        className="Refinements-closeIcon"
        onClick={[Function]}
      >
        <span
          className="screen-reader-text"
        >
          Close filters modal
        </span>
      </button>
      <button
        className="Refinements-clearButton"
        disabled={true}
        onClick={[Function]}
      >
        Clear all
      </button>
    </div>
    <Connect(RefinementList)
      handleCollapseAll={[Function]}
    />
    <EnhancedMessage
      duration={false}
      isError={false}
      isFromFilter={true}
      isVisible={true}
      message="There are no products matching your current search criteria."
      onTransitionComplete={[Function]}
      renderChildrenOn="bottom"
      showOnce={false}
    />
    <div
      className="Refinements-applyButtonWrapper"
    >
      <Button
        className="Refinements-applyButton"
        clickHandler={[Function]}
        isDisabled={true}
      >
        Apply
      </Button>
    </div>
  </div>
</div>
`;
